core/CliMulti/Process.php

```diff
@@ -288,6 +288,11 @@ private static function returnsSuccessCode($command)
     {
         $exec = $command . ' > /dev/null 2>&1; echo $?';
         $returnCode = @shell_exec($exec);
+
+        if (false === $returnCode || null === $returnCode) {
+            return false;
```

Comment:
It just came to my mind that this will actually change the current behavior. But maybe this might actually fix some possible issues.
`shell_exec` returns `false` or `null` if something didn't work with the command, otherwise the return code is returned.
If an error occurred and e.g. `null` or `false` was returned, the `trim` triggered a warning on PHP 7 and maybe even PHP 8. The error should only occur on PHP 8.1.
After the trim `$returnCode` is an empty string in case of an error and `0 == (int) $returnCode` actually evaluates to `true` and we assume the command succeeded, which it actually didn't.

Reply:
@sgiehl not sure I understand it changes behaviour. Looks like this maybe worked before? https://3v4l.org/7C8Qf

Looks generally good though 👍

Reply:
previously if `$returnCode` was `false` or `null`, this function would return `true`, now it returns `false`.

Reply:
I think this is ok to merge. This change in behaviour would almost certainly be better if it affected anyone at all